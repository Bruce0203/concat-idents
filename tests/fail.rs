use concat_idents::concat_idents;

concat_idents!(fn_name = 1, foo, bar, {
    fn fn_name() {
        /* integers are not allowed at the beginning of an ident */
    }
});

concat_idents!(fn_name = true {
    fn fn_name() {
        /* booleans are reserved key-words */
    }
});

concat_idents!(fn_name = struct {
    fn fn_name() {
        /* reserved key-words in general cannot be used as idents */
    }
});

concat_idents!(fn_name = 1 {
    fn fn_name() {
        /* an ident cannot be made of just an integer */
    }
});

concat_idents!(fn_name = foo, 1.0 {
    fn fn_name() {
        /* floats are not allowed in idents */
    }
});

concat_idents!(fn_name = foo, "bar" {
    fn fn_name() {
        /*
        string slices are not allowed to not mislead users to use this macro dynamically
        (at runtime) (would not be possible)
        */
    }
});

concat_idents!(fn_name =  {
    fn fn_name() {
        /* idents cannot be nothing */
    }
});

fn main() {}
