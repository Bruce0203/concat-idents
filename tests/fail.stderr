error: Expected ident
 --> $DIR/fail.rs:3:26
  |
3 | concat_idents!(fn_name = 1, foo, bar, {
  |                          ^

error: Expected ident
 --> $DIR/fail.rs:9:26
  |
9 | concat_idents!(fn_name = true {
  |                          ^^^^

error: Expected ident
  --> $DIR/fail.rs:15:26
   |
15 | concat_idents!(fn_name = struct {
   |                          ^^^^^^

error: Expected ident
  --> $DIR/fail.rs:21:26
   |
21 | concat_idents!(fn_name = 1 {
   |                          ^

error: Expected ident
  --> $DIR/fail.rs:27:31
   |
27 | concat_idents!(fn_name = foo, 1.0 {
   |                               ^^^

error: Expected ident
  --> $DIR/fail.rs:33:31
   |
33 | concat_idents!(fn_name = foo, "bar" {
   |                               ^^^^^

error: Expected ident
  --> $DIR/fail.rs:42:27
   |
42 | concat_idents!(fn_name =  {
   |                           ^
